docs/reference/commandline/version.md

```diff
@@ -18,38 +18,97 @@ Options:
 
 ## Description
 
-By default, this will render all version information in an easy to read
-layout. If a format is specified, the given template will be executed instead.
+The `version` command prints detailed information about the version of the
+`docker` CLI, the Docker Engine (`dockerd`), as well as other components.
 
-Go's [text/template](https://golang.org/pkg/text/template/) package
-describes all the details of the format.
+By default, this renders all version information in an easy to read
+layout. Use the [`--format`](#format) option to customize the output with a Go
+template.
 
-## Examples
+> **Note**
+>
+> The `docker` CLI also provides a `--version` flag. Unlike the `version` _command_,
+> the `--version` flag only prints version information of the `docker` CLI itself,
+> not for the Docker Engine and other components.
 
 ### Default output
 
+The default output renders all version information in an easy to read layout,
+divided into two sections; the `Client` section contains information about the
+`docker` CLI, and the `Server` section contains information about the Docker
+Engine (`dockerd`) and components used by the Engine, such as the `containerd`
+and `runc` OCI Runtimes.
+
+The information shown may differ depending on how you installed Docker and
+what components are in use. The following example shows the output of the `version`
+command on a macOS machine running Docker Desktop:
+
 ```console
 $ docker version
 
 Client:
- Version:           19.03.8
- API version:       1.40
- Go version:        go1.12.17
- Git commit:        afacb8b
- Built:             Wed Mar 11 01:21:11 2020
+ Version:           20.10.14
+ API version:       1.41
+ Go version:        go1.16.15
+ Git commit:        a224086
+ Built:             Thu Mar 24 01:49:20 2022
  OS/Arch:           darwin/amd64
  Context:           default
- Experimental:      true
 
-Server:
+Server: Docker Desktop 4.7.0 (77141)
+ Engine:
+  Version:          20.10.14
+  API version:      1.41 (minimum version 1.12)
+  Go version:       go1.16.15
+  Git commit:       87a90dc
+  Built:            Thu Mar 24 01:46:14 2022
+  OS/Arch:          linux/amd64
+ containerd:
+  Version:          1.5.11
+  GitCommit:        3df54a852345ae127d1fa3092b95168e4a88e2f8
+ runc:
+  Version:          1.0.3
+  GitCommit:        v1.0.3-0-gf46b6ba
+ docker-init:
+  Version:          0.19.0
+  GitCommit:        de40ad0
+```
+
+### Client and server versions
+
+In most cases, the `docker` CLI is used to control Docker Engine running on your
+local machine (for example, the Docker Engine running in Docker Desktop).
+Docker uses a client/server architecture, which allows you to use the `docker`
+CLI on your local machine to control a Docker Engine running on a remote machine,
+which can be (for example) a machine running in the Cloud or inside a Virtual Machine.
+
+The following example switches the `docker` CLI to use a [context](context.md)
```

Comment:
Bit on the fence on how to refer to the CLI; `docker` (the command) CLI, or if we should use "Docker CLI" (guess at some point we should "pick one".

Reply:
IMH I think the CLI tool should be referred in full for disambiguation ("Docker CLI" also instead of "docker CLI") and the command whenever possible formatted as code like you show above and sometimes as "docker command" for example, when explaining samples.

Reply:
I've seen tutorials referring it to as the Docker Engine CLI, it's not shorter but I do like how the term provides clarification and room to refer to subset of commands as extensions to the Docker Engine CLI. Something like "with version 2 of Compose, Docker extends the Engine CLI by adding the Compose CLI, a family of commands with all the functionality provided by the docker-compose binary.)".
If we consider each of these extensions each as a CLI in its own terms, then in a sense they are all Docker CLIs (Compose, Build, etc.). 
So in listing the commands in a Ref guide (thinking mostly of navigation) I'd maybe not advocate for any distinctions but explain in the chapter's intro all of these nuances of what is available in what context and why.

Reply:
> I've seen tutorials referring it to as the Docker Engine CLI

Yes, so with (e.g.) "Compose" and "Buildx" all being part of the same CLI (but in  some cases through a plugin, but this should mostly be an implementation detail), I think we should avoid _Engine_ CLI, and (unless ambiguous inn a specific context), use "Docker CLI".

Reply:
Sure, agreed. 
Thinking of communicating with users, do you think there is there the need to clarify what we call the whole vs. some parts (from previous comments I understood there was some level of uncertainty)? Or is it really just an implementation detail?
I wondering if we need that  i.e. for a set of designations to refer clearly to the:
1) whole (all docker commands);
2) (engine CLI?) set of base commands before any integrations;
3) "families" of docker commands or CLI plugins (compose, build, or others) whenever they actually can be obtained and used stand-alone in contexts other than a) Docker CLI b) Docker Desktop.